Dockerized rector does not seem to work properly

# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | Rector 0.8.x-dev              |
| Installed as   | docker container                      |

When I run the following command:
```
docker run -v $(pwd):/project rector/rector:latest process /project/src --set symfony40 --dry-run
```
or 
```
docker run -v $(pwd):/project rector/rector:latest process /project/src --set symfony40 --dry-run --autoload-file /project/vendor/autoload.php
```
https://github.com/rectorphp/rector#run-rector-in-docker

I get a lot of the following errors. e.g.:
```
 [ERROR] Could not process                                                      
         "../project/src/User/UserService.php" file, due to:  
         "Analyze error: "Class                                                 
         User/User not found.". Include 
         your files in "parameters > autoload_paths".                           
         See https://github.com/rectorphp/rector#extra-autoloading".  
```

The code itself works, so autoloading is set up correctly.

Am I doing something wrong? or is the docker container not working properly?

Hi, i was able to reproduce the error, without using `--autoload-file` option:

```
❯ docker run -v $(pwd):/project rector/rector:latest process /project/projects/car-audit/src/App/Controller/GetCarAuditStructureController.php --set symfony40 --dry-run 
Rector dev-master@c4cedab

 0/4 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%


 4/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% [ERROR] Could not process                                                      
         "../project/projects/car-audit/src/App/Controller/GetCarAuditStructureC
         ontroller.php" file, due to:                                           
         "Analyze error: "Class                                                 
         Carvago\CarAudit\App\Controller\GetCarAuditStructureController was not 
         found while trying to analyse it - discovering symbols is probably not 
         configured properly.". Include your files in "parameters >             
         autoload_paths".                                                       
         See https://github.com/rectorphp/rector#configuration".                
```

With `--autoload-file /project/vendor/autoload.php` the problem is fixed:
```
❯ docker run -v $(pwd):/project rector/rector:latest process /project/projects/car-audit/src/App/Controller/GetCarAuditStructureController.php --set symfony40 --dry-run --autoload-file /project/vendor/autoload.php
Rector v0.8.7@869441e

 0/4 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
 2/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░]  50%
 4/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 [OK] Rector is done!   
```

So i am not able to reproduce the error exactly the way you do 😞 

Please is it possible create github repository with minimal code that will break?

All you would need is just composer.json with autoloading and `src/SomeClass.php` with just `class SomeClass { }` and trying to analyse it if it works?

Now from my point of view, there might be broken autoloading in your project and might not be issue with rector in docker at all. In order to fix the issue i must be able to reproduce it, which so far i was not successful.

--- a/Dockerfile
+++ b/Dockerfile
@@ -41,6 +41,10 @@ RUN bin/rector list
 
 ENTRYPOINT [ "bin/rector" ]
 
+ENV PATH /rector/bin:$PATH
+
+VOLUME ["/project"]
+WORKDIR "/project"
 
 ## Used for getrector.org/demo
 FROM rector as rector-secured

--- a/Dockerfile
+++ b/Dockerfile
@@ -39,7 +39,7 @@ COPY . .
 # To warmup opcache a little
 RUN bin/rector list
 
-ENTRYPOINT [ "bin/rector" ]
+ENTRYPOINT [ "rector" ]
 
 ENV PATH /rector/bin:$PATH
 

--- a/README.md
+++ b/README.md
@@ -348,20 +348,20 @@ See [the contribution guide](/CONTRIBUTING.md).
 
 ## Run Rector in Docker
 
-You can run Rector on your project using Docker:
+You can run Rector on your project using Docker.
+To make sure you are running latest version, use `docker pull rector/rector`.
+
+*Note that Rector inside Docker expects your application in `/project` directory - it is mounted via volume from the current directory (`$pwd`).*
 
 ```bash
-docker run --rm -v $(pwd):/project rector/rector:latest process /project/src --set symfony40 --dry-run
-
-# Note that a volume is mounted from `pwd` (the current directory) into `/project` which can be accessed later.
+docker run --rm -v $(pwd):/project rector/rector:latest process src --set symfony40 --dry-run
 ```
 
-Using `rector.php`:
+Using `rector.php` config:
 
 ```bash
-docker run --rm -v $(pwd):/project rector/rector:latest process /project/app \
-    --config /project/rector.php \
-    --autoload-file /project/vendor/autoload.php \
+docker run --rm -v $(pwd):/project rector/rector:latest process src \
+    --config rector.php \
     --dry-run
 ```
 

--- a/rules/symfony/src/Rector/MethodCall/StringFormTypeToClassRector.php
+++ b/rules/symfony/src/Rector/MethodCall/StringFormTypeToClassRector.php
@@ -21,6 +21,7 @@ final class StringFormTypeToClassRector extends AbstractFormAddRector
      * @var string
      */
     private const DESCRIPTION = 'Turns string Form Type references to their CONSTANT alternatives in FormTypes in Form in Symfony. To enable custom types, add link to your container XML dump in "$parameters->set(Option::SYMFONY_CONTAINER_XML_PATH_PARAMETER, ...);"';
+
     public function getDefinition(): RectorDefinition
     {
         return new RectorDefinition(self::DESCRIPTION, [

--- a/rules/symfony/src/Rector/MethodCall/StringFormTypeToClassRector.php
+++ b/rules/symfony/src/Rector/MethodCall/StringFormTypeToClassRector.php
@@ -17,11 +17,13 @@ use Rector\Core\RectorDefinition\RectorDefinition;
  */
 final class StringFormTypeToClassRector extends AbstractFormAddRector
 {
+    /**
+     * @var string
+     */
+    private const DESCRIPTION = 'Turns string Form Type references to their CONSTANT alternatives in FormTypes in Form in Symfony. To enable custom types, add link to your container XML dump in "$parameters->set(Option::SYMFONY_CONTAINER_XML_PATH_PARAMETER, ...);"';
     public function getDefinition(): RectorDefinition
     {
-        $description = 'Turns string Form Type references to their CONSTANT alternatives in FormTypes in Form in Symfony. To enable custom types, add link to your container XML dump in "$parameters->set(Option::SYMFONY_CONTAINER_XML_PATH_PARAMETER, ...);"';
-
-        return new RectorDefinition($description, [
+        return new RectorDefinition(self::DESCRIPTION, [
             new CodeSample(
 <<<'CODE_SAMPLE'
 $formBuilder = new Symfony\Component\Form\FormBuilder;

